st2actions/st2actions/policies/concurrency.py

```diff
@@ -0,0 +1,94 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from st2common.constants import action as action_constants
+from st2common import log as logging
+from st2common.persistence import action as action_access
+from st2common.policies import base
+from st2common.services import coordination
+from st2common.services import executions
+from st2common.util import action_db as action_utils
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ConcurrencyPolicy(base.ResourcePolicy):
+
+    def __init__(self, policy_ref, policy_type, *args, **kwargs):
+        self.coordinator = coordination.get_coordinator()
+        self.threshold = 0
```

Comment:
I see this happens in base.ResourcePolicy. I find that pretty unintuitive to follow - how about explicitly assigning here so that there is no magic.